### DISCUSSION TITLE
assertion 'dim[0] > 0 && dim[1] == 1' failed

### DISCUSSION THREAD

**Discussion by alex-bronia** [nnstreamer/nnstreamer#4510] [Q&A]
Hello! I have a model that outputs a tensor with shape `(2,)` (Softmax probabilities of two classes). Its output is fed to a `tensor_decoder mode=image_labeling option1={LABEL_TEXT_FILE}`. I get those errors:

```
** (python3:31280): CRITICAL **: 15:46:43.233: il_getOutCaps: assertion 'dim[0] > 0 && dim[1] == 1' failed

** (python3:31280): CRITICAL **: 15:46:43.233: il_getOutCaps: assertion 'dim[0] > 0 && dim[1] == 1' failed

** (python3:31280): CRITICAL **: 15:46:43.233: il_getOutCaps: assertion 'dim[0] > 0 && dim[1] == 1' failed

** (python3:31280): CRITICAL **: 15:46:43.233: il_getOutCaps: assertion 'dim[0] > 0 && dim[1] == 1' failed

** (python3:31280): WARNING **: 15:46:43.252: Invalid param, dimensions (0) tensors (1)
```

What does `tensor_decode` expect exactly? Or is the error somewhere else, for example in ` tensor_filter framework=tensorflow2-lite model={MODEL_PATH}`? I can't seem to understand solely by log messages + documentation.

If `tensor_decode` expects a shape of `(batch, 2)` instead, how can i reshape the tensor, adding a dummy first dimension?

**Comment by alex-bronia** [comment]
Okay update: I figured out it my last idea was correct and `tensor_decode` indeed expects a shape of e.g. `(1, 2)` instead of just `(2,`). Is there a way to reshape the model output tensor to `[1, -1]` via some pipeline element? As far as I can see `tensor_transform` only allows swapping dimensions around but not actual reshaping.

**Comment by myungjoo** [comment]
@niley7464 It is probably related with the rank-limit updates, which allowed (N, 1) == (N, 0). Please take a look.
